File: app/src/main/java/com/example/bervianto/ecceg_rsa_app/ecc/Pair.java
package com.example.bervianto.ecceg_rsa_app.ecc;

import androidx.annotation.NonNull;

public class Pair<L, R> {
    public L left;
    public R right;

    public Pair(L left, R right) {
        this.left = left;
        this.right = right;
    }

    @Override
    public int hashCode() {
        return left.hashCode() ^ right.hashCode();
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Pair)) return false;
        Pair pair = (Pair) o;
        return left.equals(pair.left) && right.equals(pair.right);
    }

    @NonNull
    @Override
    public String toString() {
        return "<" + left.toString() + ", " + right.toString() + ">";
    }
}

File: app/src/main/java/id/my/berviantoleo/ecceg_rsa_app/lib/ecc/Pair.kt
package id.my.berviantoleo.ecceg_rsa_app.lib.ecc

class Pair<L, R>(var left: L?, var right: R?) {
    override fun hashCode(): Int {
        return left.hashCode() xor right.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (other !is Pair<*, *>) return false
        val pair = other
        return left == pair.left && right == pair.right
    }

    override fun toString(): String {
        return "<$left, $right>"
    }
}